torchgeo/datasets/chabud.py

```diff
@@ -0,0 +1,294 @@
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` or ``bands`` argument are invalid
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert split in self.folds
+        for band in bands:
+            assert band in self.all_bands
+
+        self.root = root
+        self.split = split
+        self.bands = bands
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        self.filepath = os.path.join(root, self.filename)
+        self.band_indices = [self.all_bands.index(b) for b in bands]
+
+        self._verify()
+
+        try:
+            import h5py  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "h5py is not installed and is required to use this dataset"
+            )
+
+        self.uuids = self._load_uuids()
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            sample containing image and mask
+        """
+        image = self._load_image(index)
+        mask = self._load_target(index)
+
+        sample = {"image": image, "mask": mask}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.uuids)
+
+    def _load_uuids(self) -> list[str]:
+        """Return the image uuids for the given split.
+
+        Returns:
+            the image uuids
+        """
+        import h5py
+
+        uuids = []
+        with h5py.File(self.filepath, "r") as f:
+            for k, v in f.items():
+                if v.attrs["fold"] in self.folds[self.split] and "pre_fire" in v:
+                    uuids.append(k)
+
+        uuids = sorted(uuids)
+        return uuids
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load a single image.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the image
+        """
+        import h5py
+
+        uuid = self.uuids[index]
+        with h5py.File(self.filepath, "r") as f:
+            pre_array = f[uuid]["pre_fire"][:]
+            post_array = f[uuid]["post_fire"][:]
+
+        # index specified bands and concatenate
+        pre_array = np.take(pre_array, indices=self.band_indices, axis=-1)
+        post_array = np.take(post_array, indices=self.band_indices, axis=-1)
```

Comment:
Didn't know about `np.take`, very fancy! Maybe this also works?
```python
pre_array = pre_array[..., self.band_indices]
```

Reply:
I don't remember why I used take. Probably was feeling fancy